app/controllers/server.rb

```diff
@@ -1,9 +1,40 @@
 module RushHour
   class Server < Sinatra::Base
+    set :show_exceptions, :after_handler
+
     not_found do
       erb :error
     end
-    
-    # Write the rest of your controller code here!
+
+    post '/sources' do
+      ValidateParams.source_response(params)
+    end
+
+    post '/sources/:IDENTIFIER/data' do
+      ValidateParams.identifier_data_response(params)
+    end
+
+    get '/sources/:IDENTIFIER' do
+      @client = ControllerLogic.find_client(params)
+      response = ValidateParams.identifier_response(params)
+      status  response[0]
+      body    response[1]
+      erb     response[-1]
```

Comment:
i believe returning an erb as your response will automatically return a status of 200 so you don't need to explicitly set the status and body. i know you might be using the body as a variable in your view template, but i'd rather you set an instance variable and pass it that way (i believe sal spoke with you about this in your evaluation)